src/libraries/Common/src/System/Security/Cryptography/RSAAndroid.cs

```diff
@@ -381,6 +381,53 @@ public override void ImportParameters(RSAParameters parameters)
                 ValidateParameters(ref parameters);
                 ThrowIfDisposed();
 
+                if (parameters.Exponent == null || parameters.Modulus == null)
+                {
+                    throw new CryptographicException(SR.Cryptography_InvalidRsaParameters);
+                }
+
+                if (parameters.D == null)
+                {
+                    if (parameters.P != null ||
+                        parameters.DP != null ||
+                        parameters.Q != null ||
+                        parameters.DQ != null ||
+                        parameters.InverseQ != null)
+                    {
+                        throw new CryptographicException(SR.Cryptography_InvalidRsaParameters);
+                    }
+                }
+                else
+                {
+                    if (parameters.P == null ||
+                        parameters.DP == null ||
+                        parameters.Q == null ||
+                        parameters.DQ == null ||
+                        parameters.InverseQ == null)
+                    {
+                        throw new CryptographicException(SR.Cryptography_InvalidRsaParameters);
+                    }
+
+                    // Half, rounded up.
+                    int halfModulusLength = (parameters.Modulus.Length + 1) / 2;
+
+                    // The same checks are done by RSACryptoServiceProvider on import (when building the key blob)
+                    // Historically RSACng let CNG handle this (reporting NTE_NOT_SUPPORTED), but on RS1 CNG let the
+                    // import succeed, then on private key use (e.g. signing) it would report NTE_INVALID_PARAMETER.
+                    //
+                    // Doing the check here prevents the state in RS1 where the Import succeeds, but corrupts the key,
+                    // and makes for a friendlier exception message.
```

Comment:
I'd replace all of this with something like

```suggestion
                    // Matching the .NET Framework RSACryptoServiceProvider behavior, as that's the .NET de facto standard
```

Since Android doesn't have the "the OS changed out from under us" that this comment is saying (and even if it does, it's not Windows CNG :smile:)